assembly/assembly-ide-war/src/main/webapp/_app/IDE.html

```diff
@@ -54,12 +54,278 @@
             }
         };
 
+        /*
+         * Determines whether string starts with specified prefix
+         */
+        if (!String.prototype.startsWith) {
+            String.prototype.startsWith = function (prefix) {
+                return !this.indexOf(prefix);
+            }
+        }
+
+        /*
+         * Determines whether string ends with specified suffix
+         */
+        if (!String.prototype.endsWith) {
+            String.prototype.endsWith = function (suffix) {
+                return this.indexOf(suffix, this.length - suffix.length) !== -1;
+            }
+        }
+
+        var Loader = new function() {
+
+            /*
+             * Show loader and load compilation-mapping.txt file to determine which IDE JavaScript file will be loaded
+             */
+            this.startLoading = function() {
+                setTimeout(function() {
+                    document.getElementById("ide-loader").style.opacity = 1;
+                }, 1);
+
+                var msg = "Cannot load compilation mappings";
+
+                try {
+                    var request = new XMLHttpRequest();
+
+                    request.onerror = request.abort = function() {
+                        console.error(msg);
+                    };
+
+                    request.onload = function() {
+                        if (request.status == 200) {
+                            Loader.determineIDEJavaScriptFileName(request.responseText);
+                        } else {
+                            console.error(msg);
+                        }
+                    };
+
+                    var url = "/_app/compilation-mappings.txt?" + new Date().getTime();
+                    request.open("GET", url, true);
+                    request.send();
+                } catch (e) {
+                    console.error(msg, e);
+                }
+            };
+
+            /*
+             * Returns browser identifier for searching in compilation-mappings.txt
+             *
+             * Many years ago this method was copied from
+             * https://code.google.com/p/google-web-toolkit/source/browse/trunk/user/src/com/google/gwt/user/rebind/UserAgentPropertyGenerator.java?r=9701
+             */
+            this.getBrowserIdentifier = function() {
+                var userAgent = navigator.userAgent.toLowerCase();
+
+                var makeVersion = function(result) {
+                    return (parseInt(result[1]) * 1000) + parseInt(result[2]);
+                };
+
+                if (userAgent.indexOf('opera') != -1) {
```

Comment:
you already add endsWith and startsWith so I think you should use includes
https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/String/includes

Reply:
Done. Thanks